mautic/mautic
# Form file field token as url

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added two new enhacement to **file field type**:
- set download link to token (for example Send form results action)
- allow match any contact field with download file url

Added new service FieldValueTransformer and unit tests.

#### Steps to test this PR:
1.  Create form with email and file field, add contact field matching

![image](https://user-images.githubusercontent.com/462477/40277760-de058554-5c24-11e8-8f12-cf3279b92dcc.png)

2.  Create Send form results action

![image](https://user-images.githubusercontent.com/462477/40277772-f4f748ce-5c24-11e8-8747-a4efb33df9d5.png)

3. Test form, upload file and send results to your email

4. See if file tokens was replaced by download url of file

5. See If contact match field was replaced by download url of uploaded file

Email before:

![image](https://user-images.githubusercontent.com/462477/40277738-a859ab06-5c24-11e8-809b-db191a27c95d.png)

After:

![image](https://user-images.githubusercontent.com/462477/40277754-c8a0e1ae-5c24-11e8-8f10-adb403fcc933.png)
